lib/srv/db/elasticsearch/engine.go

```diff
@@ -329,25 +207,31 @@ func (e *Engine) emitAuditEvent(req *http.Request, body []byte) {
 	// - we may not support given content encoding
 	query := req.URL.Query().Get("q")
 	if query == "" {
-		query = e.getQueryFromRequestBody(contentType, body)
+		query = GetQueryFromRequestBody(e.EngineConfig, contentType, body)
+	}
+
+	jsonBody, err := libaws.UnmarshalRequestBody(req)
+	if err != nil {
+		e.Log.WithError(err).Warn("Failed to read request body as JSON, omitting the json body from the audit event.")
 	}
 
 	ev := &apievents.ElasticsearchRequest{
 		Metadata: common.MakeEventMetadata(e.sessionCtx,
 			events.DatabaseSessionElasticsearchRequestEvent,
-			events.ElasticsearchRequestCode),
+			eventCode),
 		UserMetadata:     common.MakeUserMetadata(e.sessionCtx),
 		SessionMetadata:  common.MakeSessionMetadata(e.sessionCtx),
 		DatabaseMetadata: common.MakeDatabaseMetadata(e.sessionCtx),
+		StatusCode:       statusCode,
 		Method:           req.Method,
 		Path:             req.URL.Path,
 		RawQuery:         req.URL.RawQuery,
 		Body:             body,
+		JsonBody:         jsonBody,
```

Comment:
Do we need to set both body and jsonbody this will almost double the event size ? 

For me the body member filed is enough  where the event consumer can always  implement own Unmarshal function and transform base64 body to json  based on the Header on the client side.

Reply:
Yes, it would approximately double the event size. 

We could add custom processing to events to parse the body as json on the client side and avoid that overhead, and I'm inclined to go that way even if this won't be done in the immediate future.